### PR TITLE
Add new TaskOption overload for skipping input param

### PR DESCRIPTION
**Pull request by jviau**
Adds some overloads to protect against accidentally supplying `TaskOptions` as `input`.

### CODE REVIEW

File: samples/AzureFunctionsApp/AzureFunctionsApp.csproj [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
+
```
Comment (davidmrdavid) [inline samples/AzureFunctionsApp/AzureFunctionsApp.csproj:2]: nitpick: should we remove this empty change?

File: src/Abstractions/TaskOrchestrationContext.cs [hunk 2]
```diff
@@ -108,15 +108,30 @@ public abstract class TaskOrchestrationContext
...
-        return this.CallActivityAsync<object>(name, input, options);
-    }
+        => this.CallActivityAsync<object>(name, input, options);
+
+    /// <returns>
+    /// A task that completes when the activity completes or fails. The result of the task is the activity's return value.
+    /// </returns>
+    /// <inheritdoc cref="CallActivityAsync(TaskName, object?, TaskOptions?)"/>
+    public virtual Task CallActivityAsync(TaskName name, TaskOptions options)
+        => this.CallActivityAsync(name, null, options);
```
Comment (davidmrdavid) [inline src/Abstractions/TaskOrchestrationContext.cs:118]: Should we really inherit the docs from the `CallActivityAsync` with the input parameter?

Comment (davidmrdavid) [reply src/Abstractions/TaskOrchestrationContext.cs:118]: Same comment for the changes in CallSubOrchestrator

**Review by davidmrdavid — COMMENTED** [review]
Almost ready to approve, just a minor question and nitpick

File: src/Abstractions/TaskOrchestrationContext.cs (hunk 2 above)
Comment (jviau) [reply src/Abstractions/TaskOrchestrationContext.cs:118]: To be honest, I don't really know the answer to that. @cgillum suggested I do it on another PR so I did 🤷

File: samples/AzureFunctionsApp/AzureFunctionsApp.csproj (hunk 1 above)
Comment (jviau) [reply samples/AzureFunctionsApp/AzureFunctionsApp.csproj:2]: This is intentional - adding a new line for formatting of the document.

Comment (davidmrdavid) [reply samples/AzureFunctionsApp/AzureFunctionsApp.csproj:2]: Oh, interesting. So this is required as per some linter? Anyways, if it's intentional, it's fine

**Review by davidmrdavid — APPROVED** [review]
LGTM, although it would be good to hear from @cgillum on the `inheritdoc` question

File: src/Abstractions/TaskOrchestrationContext.cs (hunk 2 above)
Comment (davidmrdavid) [reply src/Abstractions/TaskOrchestrationContext.cs:118]: Well, even if this were off, it's a non-blocking concern

File: samples/AzureFunctionsApp/AzureFunctionsApp.csproj (hunk 1 above)
Comment (jviau) [reply samples/AzureFunctionsApp/AzureFunctionsApp.csproj:2]: Not required, my own preference.

**Pull request merged**